spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/sql/TestNamespaceSQL.java

```diff
@@ -187,27 +199,25 @@ public void testListNamespace() {
 
     List<Object[]> namespaces = sql("SHOW NAMESPACES IN %s", catalogName);
 
-    if (isHadoopCatalog) {
-      assertThat(namespaces).as("Should have 1 namespace").hasSize(1);
-      Set<String> namespaceNames =
-          namespaces.stream().map(arr -> arr[0].toString()).collect(Collectors.toSet());
-      assertThat(namespaceNames)
-          .as("Should have only db namespace")
-          .isEqualTo(ImmutableSet.of("db"));
+    if (isHadoopCatalog
+        || catalogConfig.get(ICEBERG_CATALOG_TYPE).equals(ICEBERG_CATALOG_TYPE_REST)) {
```

Comment:
why is this check for REST being added here? we're not testing with REST in this class, so I'm surprised to see this

Reply:
As well as my reply above, this line also is only for Spark 3.4, and it's no longer needed. Let me remove this as well. Thanks so much for pointing this.